app/services/contracts/service.go

```diff
@@ -28,23 +28,39 @@ import (
 
 	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	"github.com/ethereum/go-ethereum/common"
-	abi "github.com/limechain/hedera-eth-bridge-validator/app/clients/ethereum/contracts/bridge"
+	routerAbi "github.com/limechain/hedera-eth-bridge-validator/app/clients/ethereum/contracts/router"
 	log "github.com/sirupsen/logrus"
 )
 
+const (
+	nilErc20Address = "0x0000000000000000000000000000000000000000"
+)
+
 type Service struct {
-	address    common.Address
-	contract   *abi.Bridge
-	Client     client.Ethereum
-	mutex      sync.Mutex
-	members    Members
-	serviceFee ServiceFee
-	logger     *log.Entry
+	address        common.Address
+	routerContract *routerAbi.Router
+	Client         client.Ethereum
+	mutex          sync.Mutex
+	members        Members
+	serviceFee     ServiceFee
+	logger         *log.Entry
 }
 
-func (bsc *Service) IsValidBridgeAsset(tokenId string) (bool, string) {
-	// TODO: Do magic here
-	return true, config.LoadConfig().Hedera.Eth.WhbarContractAddress
+func (bsc *Service) IsValidBridgeAsset(opts *bind.CallOpts, tokenId string) (bool, string, error) {
+	asset, err := bsc.routerContract.NativeToWrappedToken(
```

Comment:
lets remove the concept of `asset`. Asset can be used for both `native` and `wrapped` tokens. Lets use the correct name for the corresponding token.
If the token is Hedera -> `nativeToken` if it is Ethereum erc20 -> `wrappedToken`